services/cd-service/pkg/repository/repository.go

```diff
@@ -1982,42 +2018,47 @@ func (s *State) WriteCurrentlyDeployed(ctx context.Context, transaction *sql.Tx,
 	if err != nil {
 		return err
 	}
-	for envNameIndex := range envNames {
-		envName := envNames[envNameIndex]
+	apps, err := s.GetApplicationsFromFile()
+	if err != nil {
+		return err
+	}
 
-		if apps, err := s.GetEnvironmentApplications(ctx, transaction, envName); err != nil {
-			return err
-		} else {
-			for _, appName := range apps {
-				var version *uint64
-				version, err = s.GetEnvironmentApplicationVersionFromManifest(envName, appName)
-				if err != nil {
-					return fmt.Errorf("could not get version of app %s in env %s", appName, envName)
-				}
-				var versionIntPtr *int64
-				if version != nil {
-					var versionInt = int64(*version)
-					versionIntPtr = &versionInt
-				} else {
-					versionIntPtr = nil
-				}
-				deployment := db.Deployment{
-					EslVersion:    0,
-					Created:       time.Time{},
-					App:           appName,
-					Env:           envName,
-					Version:       versionIntPtr,
-					TransformerID: 0,
-					Metadata: db.DeploymentMetadata{
-						DeployedByName:  "",
-						DeployedByEmail: "",
-					},
-				}
-				err = dbHandler.DBWriteDeployment(ctx, transaction, deployment, 0)
-				if err != nil {
-					return fmt.Errorf("error writing Deployment to DB for app %s in env %s: %v", deployment.App, deployment.Env, err)
-				}
+	for _, appName := range apps {
+		deploymentsForApp := map[string]int64{}
+		for _, envName := range envNames {
+			var version *uint64
+			version, err = s.GetEnvironmentApplicationVersionFromManifest(envName, appName)
+			if err != nil {
+				return fmt.Errorf("could not get version of app %s in env %s", appName, envName)
 			}
+			var versionIntPtr *int64
+			if version != nil {
+				var versionInt = int64(*version)
+				versionIntPtr = &versionInt
+				deploymentsForApp[envName] = int64(*version)
+			} else {
+				versionIntPtr = nil
+			}
+			deployment := db.Deployment{
+				EslVersion:    0,
+				Created:       time.Time{},
+				App:           appName,
+				Env:           envName,
+				Version:       versionIntPtr,
+				TransformerID: 0,
+				Metadata: db.DeploymentMetadata{
+					DeployedByName:  "",
+					DeployedByEmail: "",
+				},
+			}
+			err = dbHandler.DBWriteDeployment(ctx, transaction, deployment, 0)
+			if err != nil {
+				return fmt.Errorf("error writing Deployment to DB for app %s in env %s: %w", deployment.App, deployment.Env, err)
+			}
+		}
+		err = dbHandler.DBWriteAllDeploymentsForApp(ctx, transaction, 0, appName, deploymentsForApp)
```

Comment:
if this migration already ran, we won't execute it again, right? So we have the deployments, but not "all deployments".
 Is this an issue?

Reply:
Yup, you are correct. I separated these migrations for now